suite-common/wallet-utils/src/accountUtils.ts

```diff
@@ -1242,37 +1242,43 @@ export const prepareNewAccountPayload = async ({
     backendType,
     selectedAccount,
     accountTypes,
-    deviceState,
-    useEmptyPassphrase = true,
+    device,
 }: {
     accountType: AccountType;
     networkSymbol: NetworkSymbol;
     index: number;
     backendType?: TrezorConnectBackendType;
     selectedAccount?: NetworkAccount;
     accountTypes?: NetworkAccount[];
-    deviceState?: StaticSessionId;
-    useEmptyPassphrase?: boolean;
+    device: TrezorDevice;
 }) => {
     const networkAccount =
         selectedAccount ?? accountTypes?.find(v => v.accountType === accountType);
 
-    const newPath = networkAccount?.bip43Path.replace('i', String(index));
+    const accountIndex = index + (accountType === 'ledger' ? 1 : 0);
```

Comment:
@izmy can you please create util for the offset? it is also used in `discoverAccounts.ts` in connect

`offset: isEvmLedger(account, coinInfo) ? 1 : 0,`